### PR TITLE
Add custom events for address updating

### PR DESCRIPTION
**Pull request by robertfairhead**
Closes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/5450

Take a look at the `getAddressFailure` function in `letters.js` in particular since I refactor that bit out.

### CODE REVIEW

File: src/js/letters/containers/AddressSection.jsx [hunk 1]
```diff
@@ -142,10 +142,12 @@ export class AddressSection extends React.Component {
       // Reset all the error messages in case they go to edit again; should be pointless
       errorMessages: {}
     });
+    window.dataLayer.push({ event: 'letter-update-address-submit' });
```
Comment (markgreenburg) [inline src/js/letters/containers/AddressSection.jsx]: For consistency, you could probably throw this into the `saveAddressPending` action creator like you did with `success` and `failure`. Is there a reason why you opted to put this one in the component instead?

File: src/js/letters/containers/AddressSection.jsx [hunk 2]
```diff
@@ -142,10 +142,12 @@ export class AddressSection extends React.Component {
...
     this.props.saveAddress(this.state.editableAddress);
   }
 
   handleCancel = () => {
+    window.dataLayer.push({ event: 'letter-update-address-cancel' });
```
Comment (markgreenburg) [inline src/js/letters/containers/AddressSection.jsx]: So I guess the intent with this and `'letter-update-address-started'` is to log each time a user hits edit or cancel, understanding that there may be several of these per user? If so, 👍

File: src/js/letters/containers/AddressSection.jsx [hunk 3]
```diff
@@ -154,6 +156,11 @@ export class AddressSection extends React.Component {
     });
   }
 
+  startEditing = () => {
```
Comment (markgreenburg) [inline src/js/letters/containers/AddressSection.jsx]: 👍

**Review by markgreenburg — APPROVED** [review]
LGTM! Also checked out this branch and ran locally without any problems.

File: src/js/letters/containers/AddressSection.jsx (hunk 1 above)
Comment (robertfairhead) [reply src/js/letters/containers/AddressSection.jsx]: Yeah fair enough, pushed a commit to move it.

I added that one first and didn't know i'd be touching actions at the time. But that makes sense.

**Pull request merged**